@navikt/core/react/src/tabs/TabList.tsx

```diff
@@ -18,13 +28,110 @@ export type ListType = React.ForwardRefExoticComponent<
 >;
 
 const List = forwardRef<HTMLDivElement, ListProps>(
-  ({ className, ...rest }, ref) => (
-    <TabsList
-      {...rest}
-      ref={ref}
-      className={cl("navds-tabs__tablist", className)}
-    />
-  )
+  ({ className, ...rest }, ref) => {
+    const listRef = useRef<HTMLDivElement | null>(null);
+    const mergedRef = mergeRefs([listRef, ref]);
+    const [displayScroll, setDisplayScroll] = useState({
+      start: false,
+      end: false,
+    });
+
+    const updateScrollButtonState = useMemo(
+      () =>
+        debounce(() => {
+          if (!listRef?.current) return;
+          const { scrollWidth, clientWidth } = listRef?.current;
+          let showStartScroll;
+          let showEndScroll;
+
+          const scrollLeft = listRef?.current?.scrollLeft;
+          // use 1 for the potential rounding error with browser zooms.
+          showStartScroll = scrollLeft > 1;
+          showEndScroll = scrollLeft < scrollWidth - clientWidth - 1;
+
+          setDisplayScroll((displayScroll) =>
+            showStartScroll === displayScroll.start &&
+            showEndScroll === displayScroll.end
+              ? displayScroll
+              : { start: showStartScroll, end: showEndScroll }
+          );
+        }),
+      []
+    );
+
+    useEffect(() => {
+      const handleResize = debounce(() => {
+        updateScrollButtonState();
+      });
```

Comment:
Her trenger vi ikke lenger å wrappe `updateScrollButtonState` i debounce, for nå er den allerede debouncet på L41. Da slipper vi også `clear`-kallet på L77.

Reply:
🔨